### PR TITLE
Preferences test

### PR DESCRIPTION
**Pull request by ylavoie**
This PR validates the ability to switch language in User Preferences

### CODE REVIEW

**Review by Jankyboy — APPROVED** [review]

**Comment by ehuelsmann** [comment]
@Jankyboy could you explain why you are approving a failing PR which is marked as draft in a project where you spent exactly zero minutes building credibility and to further reduce your goodwill opened a nonsense PR without explanation and without responding to a request for explanation?

**Comment by ylavoie** [comment]
Please squash, individual commits bear little value.

File: old/lib/LedgerSMB/DBObject/User.pm [hunk 1]
```diff
@@ -72,11 +71,9 @@ sub change_my_password {
     );
     if (!$verify){
         $self->error($self->{_locale}->text('Incorrect Password'));
+        die;
```
Comment (ehuelsmann) [inline old/lib/LedgerSMB/DBObject/User.pm:74]: Why this "die"? Doesn't $self->error do that already??

File: xt/66-cucumber/01-basic/preferences.feature [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+   And I select "<selection>" from the drop down "Language"
+   And I save the page
+   And I select the "Preferences" tab
+  Then I expect "<translation>" to be selected for "<language>"
+  When I select "<default>" from the drop down "<language>"
+   And I click "<save>" to save the page
+   And I select the "Preferences" tab
+  Then I expect "English" to be selected for "Language"
+
+# Non UTF8 accented characters are confirmed working
```
Comment (ehuelsmann) [inline xt/66-cucumber/01-basic/preferences.feature:23]: No need to put this in comments: `Examples` should be able to have descriptions: https://github.com/cucumber/cucumber/blob/master/gherkin/testdata/good/descriptions.feature#L50

File: old/lib/LedgerSMB/DBObject/User.pm (hunk 1 above)
Comment (ylavoie) [reply old/lib/LedgerSMB/DBObject/User.pm]: Debugging leftover. Fixed

File: xt/66-cucumber/01-basic/preferences.feature (hunk 2 above)
Comment (ylavoie) [reply xt/66-cucumber/01-basic/preferences.feature]: Fixed

**Pull request merged**